aws_lambda_powertools/metrics/metrics.py

```diff
@@ -168,6 +178,129 @@ def clear_default_dimensions(self) -> None:
         self.default_dimensions.clear()
 
     def clear_metrics(self) -> None:
-        super().clear_metrics()
-        # re-add default dimensions
-        self.set_default_dimensions(**self.default_dimensions)
+        self.provider.clear_metrics()
+
```

Comment:
check whether tests are accurate since we're not resetting local default dimensions.

Reply:
Hi @heitorlessa! I think his comment should be on line 177 instead of here.

I added a line to clear the default dimensions in the provider as well. My understanding of this is because we are passing `default_dimensions` to the provider and we need to clear it there and locally. Our tests cover these lines, and we have tests to back it up.

I created an exploratory test and it works as expected.


```python
import json

from aws_lambda_powertools import Logger, Metrics
from aws_lambda_powertools.utilities.typing import LambdaContext
from aws_lambda_powertools.metrics import MetricUnit

metrics = Metrics(namespace="test")
metrics.set_default_dimensions(app="powertools")

def lambda_handler(event, context: LambdaContext):
    metrics.add_metric(name="metrics1", unit=MetricUnit.Count, value=1)
    metrics.clear_default_dimensions()
    metrics.flush_metrics()

    metrics.add_metric(name="metrics2", unit=MetricUnit.Count, value=1)
    metrics.flush_metrics()
    return {"sucesso": "ok"}
```

OUTPUT

```sh
{"_aws":{"Timestamp":1690907256372,"CloudWatchMetrics":[{"Namespace":"test","Dimensions":[["app","service"]],"Metrics":[{"Name":"metrics1","Unit":"Count"}]}]},"app":"powertools","service":"PowertoolsHelloWorld","metrics1":[1.0]}
{"_aws":{"Timestamp":1690907256372,"CloudWatchMetrics":[{"Namespace":"test","Dimensions":[["service"]],"Metrics":[{"Name":"metrics2","Unit":"Count"}]}]},"service":"PowertoolsHelloWorld","metrics2":[1.0]}
```


Let me know if you have any other consideration.

Reply:
It's missing the test for it only ;) Like this but with a provider being explicitly passed: https://github.com/aws-powertools/powertools-lambda-python/blob/develop/tests/functional/test_metrics.py#L1068

Reply:
ah done ;) https://github.com/aws-powertools/powertools-lambda-python/pull/2194/commits/92eda07d478f5a9c283d9d787739321cb49bdd4c 

<3